Assets/AnkrSDK/Plugins/WalletConnect.VersionShared/Utils/WalletDownloadHelper.cs

```diff
@@ -1,17 +1,29 @@
+using System;
 using System.Collections.Generic;
 using System.Linq;
-using AnkrSDK.WalletConnectSharp.Unity.Models.DeepLink;
-using AnkrSDK.WalletConnectSharp.Unity.Models.DeepLink.Helpers;
+using AnkrSDK.WalletConnect.VersionShared.Models.DeepLink;
+using AnkrSDK.WalletConnect.VersionShared.Models.DeepLink.Helpers;
 using Cysharp.Threading.Tasks;
 using Newtonsoft.Json;
 using UnityEngine;
 using UnityEngine.Networking;
 
-namespace AnkrSDK.WalletConnectSharp.Unity.Utils
+namespace AnkrSDK.WalletConnect.VersionShared.Utils
 {
 	public static class WalletDownloadHelper
 	{
-		public static async UniTask<Dictionary<string, AppEntry>> FetchWalletList(bool downloadImages)
+		public static async UniTask<WalletEntry> FindWalletEntry(Wallets wallet)
+		{
+			var supportedWallets = await WalletDownloadHelper.FetchWalletList(false);
```

Comment:
I think we should cache it (it is quite a heavy operation to fetch all of them) - and provide bool forceUpdateList = false to make it possible to invalidate cache